### DISCUSSION TITLE
Can flet be combined with webgl to write glsl?

### DISCUSSION THREAD

**Discussion by cgeffect** [flet-dev/flet#2413] [Q&A]
I want to develop a simple image post-processing web page, but I don't want to learn complex JavaScript. Can flets be embedded with webgl, and write glsl? In other words, is it possible to embed a native js module into a flet?

**Comment by 09u2h4n** [comment]
It is possible to use js in python but it is not directly possible use webgl and add to page.

What can you do then?
Well you need to use js in the backend.
* Maybe start with creating js module that uses webgl
* Then create a html page and add the module here finally start a local server
* In flet you need to use [webview](https://flet.dev/docs/controls/webview) and connect to localhost

I didn't try the algorithm above me but i will try as soon as possible

Then you can use webgl in python using js in the backend.

Or wait until someone builds webgl and flet implemention or you can use js in the flet's next versions

**Reply by 09u2h4n** [reply]
You can use [Py Mini Racer](https://github.com/sqreen/PyMiniRacer) as a executor js in python but i think it has to be compiled for pyodide
